TeamCode/src/main/java/com/acmerobotics/relicrecovery/motion/MotionProfile.java

```diff
@@ -0,0 +1,146 @@
+     * @param a the acceleration to apply
+     * @param t the time to apply it for
+     */
+    public void appendInfiniteJerkControl(double a, double t) {
+        MotionState segmentStart = new MotionState(end.x, end.v, a, 0, end.t);
+        segments.add(new MotionSegment(segmentStart, t));
+        end = segments.get(-1).end();
+    }
+
+    /**
+     * appends another profile to this one
+     * note: the segments themselves are not appended, rather the same controls are applied
+     * @param other
+     */
+    public void appendProfile(MotionProfile other) {
+        for (MotionSegment seg: other.segments) {
+            appendControl(seg.start().j, seg.dt());
+        }
+    }
+
+    public MotionState end() {
+        return end;
+    }
+
+    /**
+     * get the motion state at the specified time
+     * @return the motion state if it is contained in the profile, otherwise null
+     */
+    public MotionState get(double t) {
+        for (MotionSegment segment: segments) {
+            if(segment.start().t <= t && t <= segment.end().t) {
+                return segment.get(t);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * remove all parts of the profile before a time
+     * @param t time
+     * @return the profile with parts before t removed
+     */
+    public MotionProfile trimBefore(double t) {
+        SuperArrayList<MotionSegment> newSegments = new SuperArrayList<>();
+        for (MotionSegment segment: segments) {
+            if (segment.end().t < t) {
+                continue;
+            }
+            if (segment.start().t < t) {
+                newSegments.add(new MotionSegment(segment.get(t), segment.end().t - t));
+            } else {
+                newSegments.add(segment);
+            }
+        }
+        return new MotionProfile(newSegments);
+    }
+
+    /**
+     * remove all parts of the profile after a time
+     * @param t time
+     * @return the profile with parts after t removed
+     */
+    public MotionProfile trimAfter(double t) {
+        SuperArrayList<MotionSegment> newSegments = new SuperArrayList<>();
+        for (MotionSegment segment: segments) {
+            if (segment.start().t > t) {
+                continue;
+            }
+            if (segment.end().t > t) {
+                newSegments.add(new MotionSegment(segment.start(), t - segment.start().t));
+            } else {
+                newSegments.add(segment);
+            }
+        }
+        return new MotionProfile(newSegments);
+    }
+
+    /**
+     * find the next time the profile will be at a position
+     *
+     */
+    public double timeAtPos(double pos) {
+        for (MotionSegment seg: segments) {
+            double t = seg.timeAtPos(pos);
+            if (t != -1) return t;
+        }
+        return -1;
+    }
+
+    /**
+     * remove all redundant motion segments from the profile
+     */
+    public void dereduntantify() {
```

Comment:
Is this the best name? Thanks 254.

Reply:
no, 254 names the method consolidate
trying to not directly coppy
this name is so much more descriptive

Reply:
Then I would propose something like `removeEmptySegments` that describes exactly/unambiguously what the method does.